test/torchaudio_unittest/transforms/batch_consistency_test.py

```diff
@@ -1,222 +1,181 @@
+        self.assert_batch_consistency(transform, spec)
 
-        # Batch then transform
-        computed = torchaudio.transforms.Resample()(waveform.repeat(3, 1, 1))
+    def test_batch_Resample(self):
+        waveform = torch.randn(3, 2, 2786)
+        transform = T.Resample()
 
-        self.assertEqual(computed, expected)
+        self.assert_batch_consistency(transform, waveform)
 
     def test_batch_MelScale(self):
-        specgram = torch.randn(2, 201, 256)
-
-        # Single then transform then batch
-        expected = torchaudio.transforms.MelScale()(specgram).repeat(3, 1, 1, 1)
-
-        # Batch then transform
-        computed = torchaudio.transforms.MelScale()(specgram.repeat(3, 1, 1, 1))
+        specgram = torch.randn(3, 2, 201, 256)
+        transform = T.MelScale()
 
-        # shape = (3, 2, 128, 256)
-        self.assertEqual(computed, expected)
+        self.assert_batch_consistency(transform, specgram)
 
     def test_batch_InverseMelScale(self):
         n_mels = 32
         n_stft = 5
-        mel_spec = torch.randn(2, n_mels, 32) ** 2
-
-        # Single then transform then batch
-        expected = torchaudio.transforms.InverseMelScale(n_stft, n_mels)(mel_spec).repeat(3, 1, 1, 1)
-
-        # Batch then transform
-        computed = torchaudio.transforms.InverseMelScale(n_stft, n_mels)(mel_spec.repeat(3, 1, 1, 1))
-
-        # shape = (3, 2, n_mels, 32)
+        mel_spec = torch.randn(3, 2, n_mels, 32) ** 2
+        transform = T.InverseMelScale(n_stft, n_mels)
 
         # Because InverseMelScale runs SGD on randomly initialized values so they do not yield
         # exactly same result. For this reason, tolerance is very relaxed here.
-        self.assertEqual(computed, expected, atol=1.0, rtol=1e-5)
+        self.assert_batch_consistency(transform, mel_spec, atol=1.0, rtol=1e-5)
 
     def test_batch_compute_deltas(self):
-        specgram = torch.randn(2, 31, 2786)
+        specgram = torch.randn(3, 2, 31, 2786)
+        transform = T.ComputeDeltas()
 
-        # Single then transform then batch
-        expected = torchaudio.transforms.ComputeDeltas()(specgram).repeat(3, 1, 1, 1)
-
-        # Batch then transform
-        computed = torchaudio.transforms.ComputeDeltas()(specgram.repeat(3, 1, 1, 1))
-
-        # shape = (3, 2, 201, 1394)
-        self.assertEqual(computed, expected)
+        self.assert_batch_consistency(transform, specgram)
 
     def test_batch_mulaw(self):
-        waveform = common_utils.get_whitenoise(sample_rate=8000, duration=1, n_channels=2)
+        waveform = common_utils.get_whitenoise(sample_rate=8000, duration=1, n_channels=6)
+        waveform = waveform.reshape(3, 2, -1)
 
         # Single then transform then batch
-        waveform_encoded = torchaudio.transforms.MuLawEncoding()(waveform)
-        expected = waveform_encoded.unsqueeze(0).repeat(3, 1, 1)
+        expected = []
+        for i in range(3):
+            expected.append(T.MuLawEncoding()(waveform[i]))
```

Comment:
how about a list comprehension instead? e.g.
```suggestion
        expected = [T.MuLawEncoding()(waveform[i]) for i in range(3)]
```

Reply:
Thanks. I addressed it in https://github.com/pytorch/audio/pull/1772/commits/86d05b02bb78aa073a83785e269417b5b100fb95